src/world/area_kkj/kkj_19/AF7C00.c

```diff
@@ -4,29 +4,273 @@
+    Bytecode *args = script->ptrReadPos;
+
+    if (isInitialCall) {
+        D_80241B10_AF9460 = 0;
+    }
+
+    if (D_80241B10_AF9460 != 0) {
+        D_80241B10_AF9460 = 0;
+        evt_set_variable(script, *args, D_80241B14_AF9464);
+        return ApiStatus_DONE2;
+    }
+
+    return ApiStatus_BLOCK;
+}
+#else
 INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_802404DC_AF7E2C);
+#endif
+
+ApiStatus func_80240530_AF7E80(Evt* script, s32 isInitialCall) {
+    D_80241B14_AF9464 = evt_get_variable(script, *script->ptrReadPos);
+    D_80241B10_AF9460 = 1;
+
+    return ApiStatus_DONE2;
+}
+
+extern s32 D_802461F0[];
+
+ApiStatus func_80240568_AF7EB8(Evt* script, s32 isInitialCall) {
+    s32* ptr = (s32*)evt_get_variable(script, *script->ptrReadPos);
+    s32 i;
+
+    if (ptr != NULL) {
+        for (i = 0; ptr[i] != 0; i++) {
+            D_802461F0[i] = ptr[i];
+        }
+        D_802461F0[i] = 0;
+    } else {
+        for (i = 0; i < 112; i++) {
+            D_802461F0[i] = i + 16;
+            D_802461F0[112] = 0;
+        }
+    }
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_80240530_AF7E80);
+    return ApiStatus_DONE2;
+}
+
+extern s32 D_802463B8_kkj_19[];
+
+ApiStatus func_80240604_AF7F54(Evt* script, s32 isInitialCall) {
+    s32* ptr = (s32*) evt_get_variable(script, *script->ptrReadPos);
+    s32 i;
+
+    if (ptr != NULL) {
+        for (i = 0; ptr[i] != 0; i++) {
+            D_802463B8_kkj_19[i] = ptr[i];
+        }
+        D_802463B8_kkj_19[i] = 0;
+    } else {
+        for (i = 0; i < 91; i++) {
+            D_802463B8_kkj_19[i] = i + 128;
+            D_802463B8_kkj_19[91] = 0;
+        }
+    }
+
+    return ApiStatus_DONE2;
+}
+
+extern s32 D_80246568_C8C018[];
+
+#include "hud_element.h"
+
+void func_802406A0_AF7FF0(void) {
+    hud_element_draw_clipped(D_80246568_C8C018[0]);
+}
+
+extern s32 D_8024652C;
+extern s32 D_80246530;
+extern s32 D_80246534;
+extern s32 D_80246538_kkj_19[];
+extern s32 D_80246550_kkj_19;
+extern s32 D_80246554;
+extern u32 D_80246558_kkj_19;
+extern s32 D_8024655C;
+extern s32 D_80246560_kkj_19;
+extern s32 D_80246578[];
+extern s32 D_8024659C[];
+extern s32 D_802465A0;
+extern HudScript HES_AButton;
+extern HudScript HES_BlueMeter;
+extern HudScript HES_MashAButton;
+extern s32 dgb_08_npcGroup_80246528;
+
+ApiStatus func_802406C4_AF8014(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 hudElemID;
+    s32 temp;
+    s32* ptr;
+    s32 i;
+
+    if (isInitialCall) {
+        D_80246558_kkj_19 = 0;
+    }
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_80240568_AF7EB8);
+    switch (D_80246558_kkj_19) {
+        case 0:
+            D_80246534 = evt_get_variable(script, *args);
+            D_80246554 = create_generic_entity_frontUI(NULL, func_802406A0_AF7FF0);
+            dgb_08_npcGroup_80246528 = 0;
+            D_80246530 = 0;
+            for (i = 0, ptr=&D_8024659C[0]; i < 10; i++, ptr--) {
+                *ptr = 0;
```

Comment:
we should be able to use i as an array index here:

`D_8024659C[i]...`

Reply:
I tried but I couldn't make it work with i as the index. Here's what I'm getting:
![image](https://user-images.githubusercontent.com/112147187/187033862-7af7792b-1405-4ba4-a0ba-df677a7e5951.png)

For some reason `v1` is supposed to start pointing at `D_8024659C[0]` and then it starts decreasing (`D_8024659C[-1]`, `D_8024659C[-2]`, ...).

Reply:
the way to write this, then, is to use D_80246578. D_8024659C is a compiler generated symbol which actually marks the end of the array. Writing it as D_80246578 will give you the correct final address.

Reply:
That's clever! Fixed the index and removed `ptr`.